### PR TITLE
Dockerfile changes for python packages and api support

### PR DESCRIPTION
**Pull request by navinrathore**
Python api support
Release 0.3.4

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,9 @@ USER root
 WORKDIR /zingg-0.3.3-SNAPSHOT
 RUN curl --location https://github.com/zinggAI/zingg/releases/download/v0.3.3/zingg-0.3.3-SNAPSHOT-spark-3.1.2.tar.gz | \
 tar --extract --gzip --strip=1 
-RUN chmod -R a+rwx /zingg-0.3.3-SNAPSHOT/models
-RUN chown -R 1001 /zingg-0.3.3-SNAPSHOT/models
+RUN pip install -r python/requirements.txt
```
Comment (sonalgoyal) [inline Dockerfile:11]: will this install the zingg package as well?

File: Dockerfile [hunk 2]
```diff
@@ -8,7 +8,9 @@ USER root
 WORKDIR /zingg-0.3.3-SNAPSHOT
```
Comment (sonalgoyal) [inline Dockerfile:8]: wont these be 0.3.4?

**Review by sonalgoyal — CHANGES_REQUESTED** [review]
0.3.4 and zingg pip package

File: Dockerfile (hunk 1 above)
Comment (navinrathore) [reply Dockerfile:11]: No.
Should it be? Hardly, any advantage.

Comment (navinrathore) [reply Dockerfile:11]: no

**Comment by navinrathore** [comment]
Yes. It should be 0.3.4 only.
Will not install zingg package.
Will revert this PR.
It is due to Auto merge during rebase issue.

File: Dockerfile (hunk 1 above)
Comment (sonalgoyal) [reply Dockerfile:11]: I think we should do a pip install of zingg....not sure if it adds much, but just for the sake of complete release being in docker.

File: Dockerfile [hunk 3]
```diff
@@ -8,7 +8,9 @@ USER root
...
 RUN curl --location https://github.com/zinggAI/zingg/releases/download/v0.3.3/zingg-0.3.3-SNAPSHOT-spark-3.1.2.tar.gz | \
 tar --extract --gzip --strip=1 
-RUN chmod -R a+rwx /zingg-0.3.3-SNAPSHOT/models
-RUN chown -R 1001 /zingg-0.3.3-SNAPSHOT/models
+RUN pip install -r python/requirements.txt
+ENV PYTHONPATH /zingg-0.3.4-SNAPSHOT
```
Comment (sonalgoyal) [inline Dockerfile:12]: why do we need pythonpath?

**Review by sonalgoyal — CHANGES_REQUESTED** [review]
rm pypath
add pip install

**Pull request closed without merging**